### PR TITLE
Remove the environment variables for debug assertions

### LINKED ISSUE #14009: Add a `mach` argument to enable debug assertions
Follow up from #13387.

`mach` currently honors the `SERVO_ENABLE_DEBUG_ASSERTIONS` environment variable to enable debug assertions. To make this option more discoverable and better documented, add a `mach` flag that does the same thing.

IMO, we can also remove the code that checks for and acts on the environment variable.

Files:
- python/servo/build_commands.py
- etc/ci/buildbot_steps.yml (pass the new flag for the `linux-rel-wpt` and `linux-rel-css` builders)

Tests:
- I can't think of a good automated test, so manual testing is probably fine.

cc @emilio @Ms2ger

### PR DESCRIPTION
**Pull request by ghost**
…assertions."

fixes servo/servo#14009

This reverts commit 97d8768238cb4cf44bf5a03609a17dd6e980895c. Follow up to servo/servo#14025

---
This change is [Reviewable](https://reviewable.io/reviews/servo/saltfs/534)

### CODE REVIEW

**Review by ghost — APPROVED** [review]

File: buildbot/master/files/config/master.cfg [hunk 1]
```diff
@@ -157,10 +157,7 @@ c['builders'] = [
     DynamicServoBuilder("arm64", CROSS_SLAVES, envs.build_arm64),
     DynamicServoBuilder("linux-dev", LINUX_SLAVES, envs.build_linux),
     DynamicServoBuilder("linux-nightly", LINUX_SLAVES, envs.build_linux),
-    DynamicServoBuilder("linux-rel-css", LINUX_SLAVES,
-                        envs.build_linux_rel_debug_assert),
-    DynamicServoBuilder("linux-rel-wpt", LINUX_SLAVES,
-                        envs.build_linux_rel_debug_assert),
+    DynamicServoBuilder("linux-rel", LINUX_SLAVES, envs.build_linux),
```
Comment (Wafflespeanut) [inline buildbot/master/files/config/master.cfg]: We don't have this builder right now. The change should be such that `linux-rel-css` and `linux-rel-wpt` take envs.build_linux for their environments.

**Review by ghost — CHANGES_REQUESTED** [review]
This shouldn't be an "exact" revert of the other PR. There's a tiny change, and we'll be good to go :)

**Comment by Wafflespeanut** [comment]
Huh, bors didn't listen to my edited r-

File: buildbot/master/files/config/master.cfg (hunk 1 above)
Comment (iamrohit7) [inline buildbot/master/files/config/master.cfg]: Added those builders and removed linux-rel

**Comment by Wafflespeanut** [comment]
The changes look good. Before we merge, please change the commit message to something meaningful like, `"Remove environment variables for debug_assertions (revert #480)"`

**Comment by KiChjang** [comment]
@Wafflespeanut Editing your message does not send another webhook event to homu, you must always remember to make a new comment instead.

**Comment by Wafflespeanut** [comment]
@KiChjang Editing comments always triggers a new webhook event (and homu does listen to it). Try r+'ing a PR, edit your message and `r=name` again, and you'll see bors say, "This PR has already been approved" :)

**Pull request merged**